IdentityCore/src/cache/crypto/MSIDAssymetricKeyKeychainGenerator.m

```diff
@@ -250,8 +252,27 @@ - (MSIDAssymetricKeyPair *)generateKeyPairForKeyDict:(NSDictionary *)attributes
         return nil;
     }
     
-    // 2. Return keys
-    MSIDAssymetricKeyPair *keyPair = [[MSIDAssymetricKeyPair alloc] initWithPrivateKey:privateKeyRef publicKey:publicKeyRef];
+    NSDictionary* publicKeyQuery = @{ (id)kSecValueRef: (__bridge id)publicKeyRef,
+     (id)kSecClass: (id)kSecClassKey,
+     (id)kSecReturnAttributes:(id)kCFBooleanTrue
+    };
+    
+    /*
+     We need this additional query because there is only one API SecKeychainItemCopyAttributesAndData
```

Comment:
Could we only run this additional query when attributes are requested? Trying to avoid performance hit for all other flows :)

Reply:
Agree. Have updated the code . Please take a look. Added lazy loading of creation time and it will be queried only once if some one calls creationTime on asymmetric key pair.